packages/@aws-cdk-testing/framework-integ/test/aws-cognito/test/integ.user-pool-group.ts

```diff
@@ -0,0 +1,37 @@
+import { App, RemovalPolicy, Stack } from 'aws-cdk-lib';
+import { IntegTest } from '@aws-cdk/integ-tests-alpha';
+import { Construct } from 'constructs';
+import { UserPool, UserPoolGroup } from 'aws-cdk-lib/aws-cognito';
+import { Role, ServicePrincipal } from 'aws-cdk-lib/aws-iam';
+
+class TestStack extends Stack {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+    const userPool = new UserPool(this, 'UserPool', {
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+
+    const role = new Role(this, 'Role', {
+      assumedBy: new ServicePrincipal('cognito-idp.amazonaws.com'),
+    });
+
+    new UserPoolGroup(this, 'UserPoolGroup', {
+      userPool: userPool,
+      groupName: 'test-group',
+      description: 'My user pool group',
+      precedence: 0,
+      role,
+    });
+
+    userPool.addGroup('another-group', {
+      precedence: 1,
+    });
+  }
+}
+
+const app = new App();
+const stack = new TestStack(app, 'user-pool-group-stack');
+
+new IntegTest(app, 'integ-user-pool-group', {
```

Comment:
Can we add any assertions on the integ test to make sure the `UserPoolGroup` works as expected?

Reply:
I tried to implement an assertion to verify that `UserGroups` are attached to the `UserPool`. 

```ts
test.assertions.awsApiCall('cognito-identity-provider', 'ListGroupsCommand', { UserPoolId: stack.userPool.userPoolId })
  .expect(ExpectedResult.objectLike({
    Groups: [
      {
        GroupName: stack.userPoolGroup.groupName,
        UserPoolId: stack.userPool.userPoolId,
        Precedence: 1,
        Description: 'My user pool group',
        RoleArn: stack.role.roleArn,
      },
      {
        GroupName: stack.anotherUserPoolGroup.groupName,
        UserPoolId: stack.userPool.userPoolId,
      },
    ],
  }));
```
However, since the order of items in the result list is non-deterministic, the integration tests sometimes fail.
I think this is problematic. 

Based on the [assertion guidelines](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md#assertions), I think assertion is not mandatory in this case. 
What do you think?

```
Some things you should look for in deciding if the test needs an assertion:

- Integrations between services (i.e. integration libraries like `aws-cdk-lib/aws-lambda-destinations`, `aws-cdk-lib/aws-stepfunctions-tasks`, etc).
- All custom resources. Must assert the expected behavior of the lambda is correct.
- Anything that bundles or deploys custom code (i.e. does a Lambda function bundled with `aws-cdk-lib/aws-lambda-nodejs` still invoke or did we break bundling behavior).
- IAM/Networking connections.
  - This one is a bit of a judgement call. Most things do not need assertions, but sometimes we handle complicated configurations involving IAM permissions or
    Networking access.
```

Reply:
@mazyu36 thank you for the detailed clarification